### ISSUE TITLE
[REQUEST] Rule-based automatic torrent management in the WebUI

### ISSUE THREAD

**Issue by j-peeters** [transmission/transmission#6710]
### What is the issue?

It would be great if we could create rules in the WebUI, based on which torrents are processed. E.g. place downloads in specific folders based on the torrents meta data/name/etc. Like in the desktop clients.

I'm on a Unraid server and I'm dearly missing this feature. For now I switched to QBittorrent. QB also doesn't have rule based management, but yout can make multiple watch folders. It works but I find that workaround a bit cumbersome and the QB interface rather overloaded.

A kind developer came up with the idea to have QB read a JSON file with rules so that it doesn't need to be build in in the webUI (yet) (https://github.com/qbittorrent/qBittorrent/issues/5779#issuecomment-1962997699).

1) I'm wondering if something like that possible in the Web version of Transmission? Set rules from a text file?
2) Are there plans to implement rule-based torrent management in the WebUi version? 

Thanks a lot for all the great work on Transmission!

### Which application of Transmission?

Web client

**Comment by killemov** [comment]
Congratulations! You already have this functionality! But ... they are in a language known as script. All the power of your OS at your command ... as long as you ask it nicely ... in script language.

**Comment by j-peeters** [comment]
@killemov that's good news. But it's probably not as intuitive as having it in Transmission's interface. I know that script languages exist but I have no idea how this would work with a headless Transmission WebUI in Linux. I'm wondering if you could tell a bit more? Maybe you know a resource or can give an example? Thanks.

**Comment by killemov** [comment]
Sure ... I made this in 2010: [Unrar and cleanup script](https://forum.transmissionbt.com/viewtopic.php?t=10364)

**Comment by j-peeters** [comment]
Thanks @killemov I unserstand the concept now but it will demand some deep study for me to be able to write something like this for organising files. I'll dive into that. 

I hope this feature request can stay alive. Would be great if this could be implemented from the UI.

**Comment by killemov** [comment]
This rule-based system you request also comes with its own language you'll need to learn. So the TM devs then need to come up with this specific language and you'll need to learn this specific language. Looks like a waste on both sides, wasted dev-time AND wasted user time for this specific system. If you learn at least bash then that will be applicable for many more things. I could see some benefit in more variables being passed to the script though.

**Comment by j-peeters** [comment]
@killemov If it's implemented how it works in Transmission on macOS it won't be a waste of time. Just enhanced usability. 
in macOS it works like this:

![20240318 Transmission macOS 1](https://github.com/transmission/transmission/assets/4502360/3088a4c6-b4ce-453c-aadb-32c67d478ba4)

![20240318 Transmission macOS 3](https://github.com/transmission/transmission/assets/4502360/8ceada38-4c00-4ce4-b053-afeb7009be01)

That's just very intuitive. I'm not sure though if the WebUI is based om the same codebase as the macOS client. But this is what I would like to see implemented.

I'll dive into bash.

**Comment by killemov** [comment]
Yes, you are right, this does look very intuitive. (Although I'd rather see the entire rule, including what to do, in a single dialog, similar to message filters in the Thunderbird mail client.) But it is also very limited if you can only specify the location(?) based on part of torrent characteristics. If the macOS also exposes a web UI, can you use this same functionality there?

**Comment by j-peeters** [comment]
I agree having everything in one dialog would even be better, never thought of that. 
You can determine the location based on more differentiators:

![20240318 Transmission 4](https://github.com/transmission/transmission/assets/4502360/e5452758-9df9-446d-8f4b-31a564e5b861)

![20240318 Transmission 5](https://github.com/transmission/transmission/assets/4502360/b582d021-e230-4d09-b281-b5243ad8c536)

Sadly the WebUI in general cannot show this feature. To me it looks the same under Linux than on macOS.

**Comment by nevack** [comment]
WebUI is the same between all platforms, AFAIK.

macOS client's grouping rules are somewhat unique to that platform, no shared code here.
